tests/unit/Query.test.ts

```diff
@@ -55,4 +56,13 @@ describe('Query', () => {
         ?.value as unknown as number) === 100
     );
   });
+
+  it('should have as many clauses as filters passed into the query', () => {
+    const searchParams = new URLSearchParams(
```

Comment:
minor nit. can we also add a sub_product facet selection here. also should validate the distinction between compound.shoulds (OR) and separate compounds (AND) - this applies when selecting different base facets

ie. `target_product=atlas & target_product>atlas>sub_product=atlas-cli & programming_language=rust` should be `(target_product = atlas OR target_product>atlas>sub_product = atlas-cli) AND (programming_language = rust)`

(which is true in this case but not really tested. the filters are:)

```
[
  {
    "compound": {
      "should": [
        { "phrase": { "query": "atlas", "path": "facets.target_product" } },
        { "phrase": { "query": "drivers", "path": "facets.target_product" } }
      ]
    }
  },
  {
    "compound": {
      "should": [
        { "phrase": { "query": "go", "path": "facets.programming_language" } }
      ]
    }
  }
]
```